doc/manipulating_images/data_preparation.rst

```diff
@@ -10,30 +10,53 @@ Data preparation: loading and basic signal extraction
 
 |
 
-.. topic:: **File names as arguments**
+.. currentmodule:: nilearn.image
 
-   Nilearn functions and objects accept file names as arguments::
+File names or image objects
+=============================
+
+File names and objects for 3D and 4D images
+-------------------------------------------
+
+All Nilearn functions accept file names as arguments::
 
     >>> from nilearn import image
     >>> smoothed_img = image.smooth_img('/home/user/t_map001.nii')  # doctest: +SKIP
 
-   Nilearn can operate on either file names or `NiftiImage objects
-   <http://nipy.org/nibabel/nibabel_images.html>`_. The later represent
-   the specified nifti files loaded in memory.
+Nilearn can operate on either file names or `NiftiImage objects
+<http://nipy.org/nibabel/nibabel_images.html>`_. The later represent the
+data loaded in memory. In the example above, the
+function :func:`smooth_img` returns a NiftiImage object, which can then
+be readily passed to other nilearn functions.
+
+In nilearn, we often use the term *"niimg"* as abbreviation that denotes
+either a file name or a `NiftiImage object
+<http://nipy.org/nibabel/nibabel_images.html>`_. 
+
+Niimgs can be 3D or 4D. A 4D niimg can be **a list of file names**, if
+these contain 3D information.
```

Comment:
I think so, I have to check. Otherwise, LGTM. You can merge it and I will fix this détail later.